src/sparseml/pytorch/torchvision/train.py

```diff
@@ -323,9 +324,20 @@ def collate_fn(batch):
         steps_per_eval = min(steps_per_eval, args.max_eval_steps)
 
     print("Creating model")
-    model = torchvision.models.get_model(
-        args.model, weights=args.weights, num_classes=num_classes
-    )
+    if args.arch_key in ModelRegistry.available_keys():
+        with torch_distributed_zero_first(args.rank if args.distributed else None):
+            model = ModelRegistry.create(
+                key=args.arch_key,
+                pretrained=args.pretrained,
+                pretrained_path=args.checkpoint_path,
+                pretrained_dataset=args.pretrained_dataset,
+                num_classes=num_classes,
+            )
+    else:
+        # fall back to torchvision
+        model = torchvision.models.__dict__[args.arch_key](
+            pretrained=args.pretrained, num_classes=num_classes
+        )
```

Comment:
We should have a check and error message that lists available NM model repo keys if torchvision fallback also fails